USAGE.md

```diff
@@ -0,0 +1,65 @@
+# SyntaxTree Usage
+
+## Utilizing SyntaxTree's API
+
+Aside from providing a formatted view of AST nodes, SyntaxTree also provides access to information on each AST node.
+
+This quick tutorial will show you some common APIs in SyntaxTree. Keep in mind this is only a fraction of methods available to use and you can find more in the [official documentation](https://ruby-syntax-tree.github.io/syntax_tree/).
+
+Alright! Let's attain the `:+` operator in the following Ruby source code:
+
+```ruby
+require "syntax_tree"
+tree = SyntaxTree.parse("puts 1+1")
+```
+
+Everything with a block of code inside of it has a list of statements represented by a `SyntaxTree::Statements` node.
+
+```ruby
+statements = tree.statements
+# => (statements (command (ident "puts") (args ((binary (int "1") :+ (int "1"))))))
+```
+
+Let's extract the first (and only) statement of our source code that is a `SyntaxTree::Command` node, representing the `puts` method call.
+
+```ruby
+puts_command = statements.body.first
+# => (command (ident "puts") (args ((binary (int "1") :+ (int "1")))))
+```
+
+Using `#child_nodes` we can get an array of child nodes for any particular `SyntaxTree::Node`. In this case, the command node's child nodes are the method name and the arguments.
+
+We are only interested in the arguments, so we can use the instance method `#arguments` to access the `Syntax::Args` node directly.
+
+```ruby
+puts_command.child_nodes
+# => [(ident "puts"), (args ((binary (int "1") :+ (int "1"))))]
+
+args = puts_command.arguments
+# => (args ((binary (int "1") :+ (int "1"))))
+```
```

Comment:
I think we should say this the other way around. The important info here is that you can access `#arguments`. `#child_nodes` is more of a helper method.

~~~md
Here, we are only interested in the arguments, so we can use the instance method `#arguments` to access the `Syntax::Args` node directly.

```ruby
args = puts_command.arguments
# => (args ((binary (int "1") :+ (int "1"))))
```

Note that we can also get an array of child nodes for any particular `SyntaxTree::Node` using `#child_nodes`. In this case, the command node's child nodes would be the method name and the arguments.

```ruby
puts_command.child_nodes
# => [(ident "puts"), (args ((binary (int "1") :+ (int "1"))))]
```
~~~